internal/serve/middleware/middleware.go

```diff
@@ -177,6 +179,70 @@ func (c cspItem) String() string {
 	return fmt.Sprintf("%s %s;", c.ContentType, strings.Join(c.Policy, " "))
 }
 
+// LoggingMiddleware is a middleware that logs requests to the logger.
+func LoggingMiddleware() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(rw http.ResponseWriter, req *http.Request) {
+			mw := mutil.WrapWriter(rw)
+
+			reqCtx := req.Context()
+			logFields := log.F{
+				"req": middleware.GetReqID(reqCtx),
+			}
+			logCtx := log.Set(reqCtx, log.Ctx(reqCtx).WithFields(logFields))
+
+			tenant, err := tenant.GetTenantFromContext(reqCtx)
+			if err != nil {
+				log.Ctx(logCtx).Info("tenant cannot be derived from context")
```

Comment:
hm, what do you think about using DEBUG-level here?

Reply:
ok, I don't have much of an opinion on this!

Reply:
A lot of requests will have no tenant, and Info may be too verbose b/c it's not something important for those requests.